cmake/adjust/template/vendor/CXX_Clang.cmake

```diff
@@ -0,0 +1,51 @@
+include_guard(GLOBAL)
+
+# Debug
```

Comment:
what about macro definitions like `NDEBUG` ?

P.S. should MSVC even define should macros from commandline instead of configuring theminto the header?

Reply:
> what about macro definitions like `NDEBUG` ?

in theory we set them in MSVC frontend profile but we could skip doing so both for this profile & Clang since CMake sets `NDEBUG` at the very beginning with `CMAKE_<LANG>_FLAGS_<CONFIG>_INIT` within some defaults (ofc, it can be overwritten) - so its not MSVC which sets it but CMake

- <https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Windows-MSVC.cmake#L504>
- <https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Windows-Clang.cmake#L117>

> instead of configuring theminto the header?

to be discussed but we already generate a set of defines which is included in nabla.h iirc

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/commit/a2a7e72f42501fe71b6f4f141b92bac4d2b8cd93 I stay at defining it myself though